infra_tooling_account/infra_tooling_account/infra_tooling_grafana_stack.py

```diff
@@ -0,0 +1,386 @@
+
+    """
+
+    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
+        """
+        Initialize the GrafanaStack.
+
+        Args:
+            scope (Construct): The CDK app or stack that this stack is a part of.
+            construct_id (str): The identifier for this stack.
+            **kwargs: Arbitrary keyword arguments. Specifically looks for:
+                - project_name (str): The name of the project. Used for naming resources. Defaults to None.
+                - stage_name (str): The name of the deployment stage. Used for naming resources. Defaults to None.
+                - settings (Tuple): Tuple containing settings. Defaults to None.
+
+        """
+        self.stage_name = kwargs.pop("stage_name", None)
+        self.project_name = kwargs.pop("project_name", None)
+        self.settings: Settings = kwargs.pop("settings", None)
+
+        super().__init__(scope, construct_id, **kwargs)
+
+        (
+            input_timestream_database_arn,
+            input_timestream_kms_key_arn,
+            input_settings_bucket_arn,
+            settings_bucket,
+        ) = self.get_common_stack_references()
+
+        (
+            grafana_vpc_id,
+            grafana_security_group_id,
+        ) = self.settings.get_grafana_mandatory_settings()
+
+        (
+            grafana_key_pair_name,
+            grafana_bitnami_image,
+            grafana_instance_type,
+        ) = self.settings.get_grafana_optional_settings()
+
+        (
+            grafana_admin_secret_name,
+            grafana_admin_secret_arn,
+        ) = self.create_grafana_admin_secret()
+
+        # Create Grafana key pair if not provided
+        if not grafana_key_pair_name:
+            grafana_key_pair_name = self.create_grafana_key_pair()
+
+        self.prepare_and_upload_grafana_configuration_files(settings_bucket)
+
+        grafana_instance = self.create_grafana_instance(
+            grafana_vpc_id=grafana_vpc_id,
+            grafana_security_group_id=grafana_security_group_id,
+            input_timestream_kms_key_arn=input_timestream_kms_key_arn,
+            input_timestream_database_arn=input_timestream_database_arn,
+            input_settings_bucket_arn=input_settings_bucket_arn,
+            grafana_admin_secret_name=grafana_admin_secret_name,
+            grafana_admin_secret_arn=grafana_admin_secret_arn,
+            grafana_bitnami_image=grafana_bitnami_image,
+            grafana_key_pair_name=grafana_key_pair_name,
+            grafana_instance_type=grafana_instance_type,
+            settings_bucket_name=settings_bucket.bucket_name,
+        )
+
+        output_grafana_instance_public_ip = CfnOutput(
+            self,
+            "GrafanaPublicIp",
+            value=grafana_instance.instance_public_ip,
+            description="The Public IP of the Grafana Instance",
+            # To sign in to Grafana, go to http://<grafana-instance-public-ip>:3000
```

Comment:
I guess, it would be even better to have Grafana URL as output (so user can just copy and paste it).
Ot maybe, expose both IP and URL from stack outputs

Reply:
thank you! agree, will adjust accordingly

Reply:
Grafana URL is in the output now. I think only URL would be enough, but let me know if any further adjustments are required